pytorch/workflows/mnist.py

```diff
@@ -0,0 +1,181 @@
+
+    def forward(self, x):
+        x = F.relu(self.conv1(x))
+        x = F.max_pool2d(x, 2, 2)
+        x = F.relu(self.conv2(x))
+        x = F.max_pool2d(x, 2, 2)
+        x = x.view(-1, 4*4*50)
+        x = F.relu(self.fc1(x))
+        x = self.fc2(x)
+        return F.log_softmax(x, dim=1)
+
+def train(model, device, train_loader, optimizer, epoch, writer, log_interval):
+    model.train()
+    for batch_idx, (data, target) in enumerate(train_loader):
+        data, target = data.to(device), target.to(device)
+        optimizer.zero_grad()
+        output = model(data)
+        loss = F.nll_loss(output, target)
+        loss.backward()
+        optimizer.step()
+        if batch_idx % log_interval == 0:
+            print('Train Epoch: {} [{}/{} ({:.0f}%)]\tloss={:.4f}'.format(
+                epoch, batch_idx * len(data), len(train_loader.dataset),
+                       100. * batch_idx / len(train_loader), loss.item()))
+            niter = epoch * len(train_loader) + batch_idx
+            writer.add_scalar('loss', loss.item(), niter)
+
+def test(model, device, test_loader, writer, epoch):
+    model.eval()
+    test_loss = 0
+    correct = 0
+    with torch.no_grad():
+        for data, target in test_loader:
+            data, target = data.to(device), target.to(device)
+            output = model(data)
+            test_loss += F.nll_loss(output, target, reduction='sum').item() # sum up batch loss
+            pred = output.max(1, keepdim=True)[1] # get the index of the max log-probability
+            correct += pred.eq(target.view_as(pred)).sum().item()
+
+    test_loss /= len(test_loader.dataset)
+    print('\naccuracy={:.4f}\n'.format(float(correct) / len(test_loader.dataset)))
+    accuracy = float(correct) / len(test_loader.dataset)
+    writer.add_scalar('accuracy', accuracy, epoch)
+    return (epoch, accuracy)
+
+def epoch_step(model, device, train_loader, test_loader, optimizer, epoch, writer, log_interval):
+    train(model, device, train_loader, optimizer, epoch, writer, log_interval)
+    return test(model, device, test_loader, writer, epoch)
+
+def should_distribute():
+    return dist.is_available() and WORLD_SIZE > 1
+
+
+def is_distributed():
+    return dist.is_available() and dist.is_initialized()
+
+@inputs(
+    no_cuda=Types.Boolean,
+    batch_size=Types.Integer,
+    test_batch_size=Types.Integer,
+    epochs=Types.Integer,
+    learning_rate=Types.Float,
+    sgd_momentum=Types.Float,
+    seed=Types.Integer,
+    log_interval=Types.Integer,
+    save_model=Types.Boolean,
+    dir=Types.String)
+@outputs(out=Types.String)
+@pytorch_task(
+    workers_count=2,
+    per_replica_cpu_request="500m",
+    per_replica_memory_request="4Gi",
+    per_replica_memory_limit="8Gi",
+    per_replica_gpu_limit="1",
+)
+def mnist_pytorch_job(workflow_params, no_cuda, batch_size, test_batch_size, epochs, learning_rate, sgd_momentum, seed, log_interval, save_model, dir, out):
+    backend_type = dist.Backend.GLOO
```

Comment:
So eventually I want this boilerplate code to be in the pytorch wrapper, can you add a TODO: simplify by abstracting the boilerplate

Reply:
Should we leave TODOs in example/boilerplate code? Maybe it's better to create issue to remember this idea?

Reply:
issue works too, either way.

Reply:
issue is great, and the TODO can point to the issue :)